mantis-runtime/src/main/java/io/mantisrx/runtime/executor/StageExecutors.java

```diff
@@ -199,22 +199,44 @@ public void stop() {}
      */
     @SuppressWarnings("unchecked")
     private static <K, T, R> Observable<Observable<R>> executeMantisGroupsInParallel(Observable<Observable<MantisGroup<K, T>>> go, Computation computation,
-                                                                                     final Context context, final boolean applyTimeoutToInners, final long timeout) {
+                                                                                     final Context context, final boolean applyTimeoutToInners, final long timeout, final int concurrency) {
         logger.info("initializing {}", computation.getClass().getCanonicalName());
         computation.init(context);
 
         // from groups to observable
         final Func2<Context, Observable<MantisGroup<K, T>>, Observable<R>> c
                 = (Func2<Context, Observable<MantisGroup<K, T>>, Observable<R>>) computation;
 
-        return
-                go
-                        .lift(new MonitorOperator<>("worker_stage_outer"))
-                        .map(observable -> c
-                                .call(context, observable
-                                        .subscribeOn(Schedulers.computation())
-                                        .lift(new MonitorOperator<MantisGroup<K, T>>("worker_stage_inner_input")))
-                                .lift(new MonitorOperator<R>("worker_stage_inner_output")));
+        if(concurrency == StageConfig.DEFAULT_STAGE_CONCURRENCY) {
```

Comment:
same logic for the GroupedObservable code path as well?

Reply:
That won't suffice as we would have to avoid the merge that happens early on in the network layer as well

Reply:
gotcha